mantidproject/mantid
# LoadILLSANS v2

**Description of work.**

This PR introduces version 2 of the `LoadILLSANS` algorithm. The changes include:
- Removal of `wavelength` property of the algorithm; it is not longer necessary, as it was intended only for certain files coming from D16 and they have been since corrected,
- Introduction of new property `LoadInstrument`, set by default to `true`, that allows to choose to not load the instrument geometry to possibly save up loading time for measurements where the instrument geometry would be the same (same distance for example),
- Move to using point data for all measurement types but TOF; for TOF the histogram remains the data type
- The X-axis unit is an index for point data, not a wavelength. Wavelength remains the unit for TOF.
- Refactoring of initialisation of workspaces and placement of geometry: e.g. placement of D16 is not anymore its own function.

For now the new loader is not used in the reduction chain (`SANSILLReduction`, `SANSILLParameterScan`), and the v1 is explicitly set.

Additionally, a bug has been found in `LoadAndMerge` where the user's selection of the loader's version would take no effect as it was communicated via a positional argument rather than key-word argument. This has been fixed.




**To test:**

1. All tests should be passing
2. Load ILL SANS data, e.g. from the unit test suite:
- check that loading works as well as for the v1
- check if the loader works even with `LoadInstrument` is set to false.



Fixes #31996 . 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
